ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/StreamingIngestClient.kt

```diff
@@ -23,10 +31,121 @@ class StreamingIngestClient(
     val engineUrl: String,
     override val tokenCredential: TokenCredential,
     override val skipSecurityChecks: Boolean = false,
+    override val clientDetails: ClientDetails? = null,
 ) :
-    KustoBaseApiClient(engineUrl, tokenCredential, skipSecurityChecks),
+    KustoBaseApiClient(
+        engineUrl,
+        tokenCredential,
+        skipSecurityChecks,
+        clientDetails,
+    ),
     IngestClient {
 
+    /** Handles multiple source types for streaming ingestion. */
+    override suspend fun submitIngestion(
+        database: String,
+        table: String,
+        sources: List<SourceInfo>,
+        format: Format,
+        ingestProperties: IngestRequestProperties?,
+    ): IngestResponse {
+        require(sources.isNotEmpty()) { "At least one source is required" }
+
+        // Streaming ingestion processes one source at a time
+        val source = sources.first()
+        val operationId = UUID.randomUUID().toString()
+
+        when (source) {
+            is BlobSourceInfo -> {
+                logger.info(
+                    "Streaming ingestion from BlobSource: ${source.blobPath}",
+                )
+                submitStreamingIngestion(
+                    database = database,
+                    table = table,
+                    // Not used for blob-based streaming
+                    data = ByteArray(0),
+                    format = format,
+                    ingestProperties = ingestProperties,
+                    blobUrl = source.blobPath,
+                )
+            }
+            is FileSourceInfo -> {
+                logger.info(
+                    "Streaming ingestion from FileSource: ${source.name}",
+                )
+                val data = source.data().readBytes()
+                submitStreamingIngestion(
+                    database = database,
+                    table = table,
+                    data = data,
+                    format = format,
+                    ingestProperties = ingestProperties,
+                    blobUrl = null,
+                )
+                source.close()
+            }
+            is StreamSourceInfo -> {
+                logger.info(
+                    "Streaming ingestion from StreamSource: ${source.name}",
+                )
+                val data = source.data().readBytes()
+                submitStreamingIngestion(
+                    database = database,
+                    table = table,
+                    data = data,
+                    format = format,
+                    ingestProperties = ingestProperties,
+                    blobUrl = null,
```

Comment:
Also, is there a reason we are passing blob URL ? Streaming ingest would not have blobURL right